### ISSUE TITLE
Syntax support for Moops in contrib

### ISSUE THREAD

**Issue by srchulo** [vim-perl/vim-perl#268]
It would be really great if [Moops](https://metacpan.org/pod/Moops) syntax could also be supported as an add on in the contrib folder! Things like class, role, etc.

**Comment by hurricup** [comment]
@srchulo just curious, isn't it declared unstable and seems abandoned (no recent updates)?

**Comment by srchulo** [comment]
Yeah, I saw the unstable part, but I found it because it was recommended in Modern Perl's 2016 Edition under the [Moose and Perl OO](http://modernperlbooks.com/books/modern_perl_2016/07-object-oriented-perl.html#TW9vc2VhbmRQZXJsT08) section. I had the impression that it was popular, but maybe I was wrong.

**Comment by hurricup** [comment]
Probably @tobyink could enlighten us about the real state

**Comment by eiro** [comment]
hello,

> section. I had the impression that it was popular, but maybe I was
> wrong.

47++ on cpan. this is probably more than an impression :)

regards
marc

**Comment by hurricup** [comment]
@eiro not so hard to favorite. Not so many reverse dependencies.

**Comment by eiro** [comment]
On Tue, Dec 18, 2018 at 09:17:35AM +0000, Alexandr Evstigneev wrote:
> @eiro not so hard to favorite. Not so many reverse dependencies.

yet: i don't know other metric on the module popularity.

regards

**Comment by petdance** [comment]
I'm fine with adding something in `contrib` so long as it's clearly marked in the file as to who the author is and what the maintenance process for it is.

**Comment by srchulo** [comment]
I'd be happy to help with this, however, I'm not really familiar with the syntax used in all of the files in `contrib`. Is there a good explanation of how these files are formed? Or is it just the examples in `contrib`?

**Comment by tobyink** [comment]
It's not abandoned. And the implementation is more unstable than the syntax, which is the kind of instability which shouldn't matter to a syntax highlighter.

**Comment by tobyink** [comment]
(I'll also note that it seems a contradiction to accuse a piece of software of being both unstable and abandoned. Abandoned software is the *most* stable!)

**Comment by srchulo** [comment]
Also, I think it's good to note that any syntax support for Moops would also mean syntax support for [Kavorka](https://metacpan.org/pod/Kavorka) would be needed.
